src/habitat.cr

```diff
@@ -88,14 +90,14 @@ class Habitat
 
     class Settings
       {% if type_with_habitat.superclass && type_with_habitat.superclass.constant(:HABITAT_SETTINGS) %}
-        {% for decl in type_with_habitat.superclass.constant(:HABITAT_SETTINGS) %}
+        {% for decl in type_with_habitat.superclass.constant(:HABITAT_SETTINGS).map{ |setting| setting[:decl] } %}
           def self.{{ decl.var }}
             ::{{ type_with_habitat.superclass }}::Settings.{{ decl.var }}
           end
         {% end %}
       {% end %}
 
-      {% for decl in type_with_habitat.constant(:HABITAT_SETTINGS) %}
+      {% for decl in type_with_habitat.constant(:HABITAT_SETTINGS).map{ |setting| setting[:decl] } %}
```

Comment:
I'm not sure, but I think you can use the shorthand block syntax here: `.map(&[:decl])`. If not that's ok, but I figured it's worth trying :)